Controller/CallbackController.php

```diff
@@ -70,64 +72,88 @@ public function urlcAction(Request $request, PaymentInstruction $instruction)
 
         if ($control !== $request->request->get('md5')) {
             $logger->error(
-                '[Dotpay - URLC] pin verification failed',
-                array(
+                '[Dotpay - URLC - {dotpayTransactionId}] pin verification failed',
+                [
                     'paymentInstructionId' => $instruction->getId(),
                     'dotpayTransactionId' => $transactionId,
                     'dotpayTransactionStatus' => $transactionStatus,
-                )
+                ]
             );
 
-            return new Response('FAIL', 500);
+            return new Response('FAIL SIGNATURE', 500);
         }
 
         if (null === $transaction = $instruction->getPendingTransaction()) {
-            // this could happen if the transaction is already validated via http redirection
-            $logger->info(
-                '[Dotpay - URLC] no pending transaction found for the payment instruction',
-                array(
+            if ($instruction->getAmount() < $instruction->getDepositedAmount()) {
+                $logger->error(
+                    '[Dotpay - URLC - {dotpayTransactionId}] unable to create new transaction, all of amount has been deposited',
+                    [
+                        'paymentInstructionId' => $instruction->getId(),
+                        'dotpayTransactionId' => $transactionId,
+                        'dotpayTransactionStatus' => $transactionStatus,
+                    ]
+                );
+
+                return new Response('FAIL, TRANSACTION IS COMPLETED', 500);
+            }
+
+            $logger->error(
+                '[Dotpay - URLC - {dotpayTransactionId}] no pending transaction found for the payment instruction, creating new one',
+                [
                     'paymentInstructionId' => $instruction->getId(),
                     'dotpayTransactionId' => $transactionId,
                     'dotpayTransactionStatus' => $transactionStatus,
-                )
+                ]
             );
 
-            return new Response('FAIL', 500);
+            $payment = $ppc->createPayment($instruction->getId(), $amount);
```

Comment:
no, we must not do that: the payment instruction is done before and we should not create a new one when receiving the callback from dotpay's server